cmd/token.go

```diff
@@ -26,6 +28,19 @@ var mockCloudLoggingServerAddress string
 
 const toolName = "gotpm"
 
+type tpmWrapper struct {
+	io.ReadWriter
+}
+
+// EventLog allows caller to call the EventLogGetter function
+// of the wrapped TPM.
+func (et tpmWrapper) EventLog() ([]byte, error) {
+	if ExternalTPM != nil {
+		return client.GetEventLog(et.ReadWriter)
+	}
+	return os.ReadFile(eventLog)
+}
+
```

Comment:
I'm sorry. I should've been clearer in my last comment. Would it make sense to have the response of `openImpl` implement `EventLogGetter` and call `os.ReadFile(eventLog)` there?

Reply:
Just to clarify. Do you mean to create a new interface type like `tpmWrapper` which holds `io.ReadWriteCloser` and `EventLogGetter`? And change `openImpl` responce `io.ReadWriteCloser` to `tpmWrapper`?

Reply:
I put down the implementation in the new commit, please let me know if it is what you are suggesting. Thanks!